### PR TITLE
Feature: manage permissions of roles and group

### PR DESCRIPTION
**Pull request by Zorin95670**

### CODE REVIEW

File: src/main/resources/db/migration/V1_7_0__new_table_access_controls_permissions.sql [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    per_id      INTEGER REFERENCES permissions(per_id) ON DELETE CASCADE NOT NULL,
+    aco_id      INTEGER REFERENCES access_controls(aco_id) ON DELETE CASCADE NOT NULL,
+    insert_date TIMESTAMP NOT NULL DEFAULT now(),
+    update_date TIMESTAMP NOT NULL DEFAULT now(),
+    UNIQUE (per_id, aco_id)
+);
+
+COMMENT ON TABLE  access_controls_permissions             IS 'This table establishes a relationship between permissions and access controls. It maps which access controls are associated with specific permissions.';
+COMMENT ON COLUMN access_controls_permissions.pac_id      IS 'Primary key, serial.';
+COMMENT ON COLUMN access_controls_permissions.per_id      IS 'This column denotes the specific permission being linked to an access control. References the per_id in the permissions table. ';
```
Comment (justinehell) [inline src/main/resources/db/migration/V1_7_0__new_table_access_controls_permissions.sql:12]: remove space between `.` and `';` at end of sentence

File: src/main/java/com/ditrit/letomodelizerapi/model/permission/PermissionDirectDTO.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * @see PermissionDTO
+ */
+@EqualsAndHashCode(callSuper = true)
+@Data
+public class PermissionDirectDTO extends PermissionDTO {
+
+    /**
+     * Indicates whether the permission relationship is direct.
+     * This field distinguishes between direct and indirect relationships in the access control tree.
+     * A direct relationship implies that the current access control is immediately link to the permission,
```
Comment (justinehell) [inline src/main/java/com/ditrit/letomodelizerapi/model/permission/PermissionDirectDTO.java:19]: is immediately linked to the permission

File: src/test/resources/features/Role.feature [hunk 3]
```diff
@@ -308,3 +308,93 @@ Feature: Role feature
...
+    When I request "/roles/[role_id]/permissions" with method "GET"
+    Then I expect "200" as status code
+    And  I extract resources from response
+    And  I expect response resources length is "1"
+    And  I expect one resource contains "id" equals to "1"
+    And  I expect one resource contains "action" equals to "ACCESS"
+    And  I expect one resource contains "entity" equals to "ADMIN"
+    And  I expect one resource contains "isDirect" equals to "true" as "boolean"
+
+    # Associate group and role
```
Comment (justinehell) [inline src/test/resources/features/Role.feature:345]: Dissociate

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/main/resources/db/migration/V1_7_0__new_table_access_controls_permissions.sql (hunk 1 above)
Comment (Zorin95670) [reply src/main/resources/db/migration/V1_7_0__new_table_access_controls_permissions.sql:12]: done

File: src/main/java/com/ditrit/letomodelizerapi/model/permission/PermissionDirectDTO.java (hunk 2 above)
Comment (Zorin95670) [reply src/main/java/com/ditrit/letomodelizerapi/model/permission/PermissionDirectDTO.java:19]: done

File: src/test/resources/features/Role.feature (hunk 3 above)
Comment (Zorin95670) [reply src/test/resources/features/Role.feature:345]: done

**Review by justinehell — APPROVED** [review]

**Pull request merged**